src/test/java/com/powsybl/openloadflow/sa/OpenSecurityAnalysisTest.java

```diff
@@ -237,7 +324,7 @@ void testFourSubstations() {
             .map(b -> new Contingency(b.getId(), new BranchContingency(b.getId())))
             .collect(Collectors.toList());
 
-        OpenSecurityAnalysis securityAnalysis = new OpenSecurityAnalysis(network, new DefaultLimitViolationDetector(),
```

Comment:
To remain coherent with the rest of the code, we have to use the provider.

Reply:
Yes I think there is still some work to adapt the provider to be more generic. I am on it.

Reply:
Indeed, the provider should be used instead as it it part of the security analysis API, unlike the OpenSecurityAnalysis itself. I also did some unit tests with the OpenSecurityAnalysis in another PR, but after some thinking I think it is misleading: people tend to copy the code from unit tests to use powsybl. You should do something similar to what was done for sensitivity analysis, with the `AbstractSensitivityAnalysisTest` class.